src/main/java/hudson/tasks/junit/JUnitResultArchiver.java

```diff
@@ -241,7 +241,7 @@ public double getHealthScaleFactor() {
      *
      * @since 1.2
      */
-    @DataBoundSetter public final void setTestDataPublishers(@Nonnull List<? extends TestDataPublisher> testDataPublishers) {
+    @DataBoundSetter public final void setTestDataPublishers(@Nonnull List<TestDataPublisher> testDataPublishers) {
         this.testDataPublishers = new DescribableList<TestDataPublisher,Descriptor<TestDataPublisher>>(Saveable.NOOP);
```

Comment:
Hmm, I wonder why this was ever a `DescribableList` to begin with, if (a) we do not allow an `owner` to be set, so you cannot trigger a project save just by editing this list; (b) we do not return ``DescribableList` from the getter, so you cannot call richer methods like `get(Class)`. Anyway not an issue in this PR.